File: shoring/flywheel/src/test/java/com/panopset/tests/transformer/StandardPackagePath.java
package com.panopset.tests.transformer;

import java.io.File;

import com.panopset.compat.Fileop;

public class StandardPackagePath {
	private final String packageName;
	private String packagePath;

	public StandardPackagePath(String packageName) {
		this.packageName = packageName;
	}

	public String getRezStr(String fileName) {
		return Fileop.readTextFile(getFile(fileName));
	}

	public File getFile(String fileName) {
		return new File(String.format("%s/%s", getPackagePath(), fileName));
	}

	protected String getPackagePath() {
		if (packagePath == null) {
			packagePath = String.format("src/test/resources/%s", packageName.replace('.', '/'));
		}
		return packagePath;
	}
}

File: code/shoring/flywheel/src/test/kotlin/com/panopset/tests/transformer/StandardPackagePath.kt
package com.panopset.tests.transformer

import com.panopset.compat.Fileop
import com.panopset.compat.Logz
import java.io.File

class StandardPackagePath(private val packageName: String) {
    var packagePath: String = String.format("src/test/resources/%s", packageName.replace('.', '/'))

    fun getRezStr(fileName: String): String {
        return Fileop.readTextFile( getFile(fileName))
    }

    fun getFile(fileName: String): File {
        return File(String.format("%s/%s", packagePath, fileName))
    }
}
